frontend/src/components/HighlightSearchText.tsx

```diff
@@ -3,68 +3,190 @@ import { css } from '@emotion/css'
+const createHighlightedSegments = (text: string, searchText: string) => {
+  if (!searchText) return [{ text, isMatch: false, key: `${text}-0-false` }]
+
+  const segments: { text: string; isMatch: boolean; key: string }[] = []
+  const searchLower = searchText.toLowerCase()
+  const textLower = text.toLowerCase()
+
+  let lastIndex = 0
+  let searchIndex = textLower.indexOf(searchLower)
+
+  while (searchIndex !== -1) {
+    // Add non-matching segment before the match
+    if (searchIndex > lastIndex) {
+      const segmentText = text.slice(lastIndex, searchIndex)
+      segments.push({
+        text: segmentText,
+        isMatch: false,
+        key: `${segmentText}-${lastIndex}-false`,
+      })
+    }
+
+    // Add matching segment using the actual found match length
+    const matchText = text.slice(searchIndex, searchIndex + searchText.length)
+    segments.push({
+      text: matchText,
+      isMatch: true,
+      key: `${matchText}-${searchIndex}-true`,
+    })
+
+    lastIndex = searchIndex + searchText.length
+    searchIndex = textLower.indexOf(searchLower, lastIndex)
+  }
+
+  // Add remaining non-matching segment
+  if (lastIndex < text.length) {
+    const segmentText = text.slice(lastIndex)
+    segments.push({
+      text: segmentText,
+      isMatch: false,
+      key: `${segmentText}-${lastIndex}-false`,
+    })
+  }
+
+  return segments
+}
+
 // render text with highlights for searched filter text
 // if text is a label like 'key=value', add a toggle button that toggles between = and !=
-// if truncate is set, also make label smaller with ellipses
+// truncation is always handled by the Truncate component
 export function HighlightSearchText(
   props: Readonly<{
     text?: string
     searchText?: string
     supportsInequality?: boolean
     toggleEquality?: () => void
-    isTruncate?: boolean
+    isLink?: boolean
   }>
 ) {
-  const { text, searchText, supportsInequality, toggleEquality, isTruncate } = props
-  const segments = getSlicedText(text, searchText)
-  if (segments.length > 1) {
-    const isTruncateLabel = isTruncate && text && text.length > MAX_LABEL_WIDTH
+  const { text, searchText, supportsInequality, toggleEquality, isLink } = props
+
+  if (!text) return null
+
+  // Handle toggle button case for inequality operators
+  if (supportsInequality && parseLabel(text).oper) {
+    return <ToggleButton label={text} toggleEquality={toggleEquality} />
+  }
+
+  // Handle link rendering (with or without search highlighting)
+  if (isLink) {
+    let content
+    if (searchText) {
+      const segments = createHighlightedSegments(text, searchText)
+      if (segments.some((seg) => seg.isMatch)) {
+        // Link with search highlighting
+        content = segments.map((segment) => (
+          <span
+            key={segment.key}
+            className={segment.isMatch ? highlightClass : linkClass}
+            data-highlight={segment.isMatch ? 'true' : undefined}
+            data-link="true"
```

Comment:
data-highlight and data-link props are just one way to inform the Truncate.tsx component that a piece of text is a link and/or contains text that will need to be highlighted.